common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -101,18 +101,19 @@ public String toString() {
       rest %= MICROS_PER_MINUTE;
       if (rest != 0) {
         String s = BigDecimal.valueOf(rest, 6).stripTrailingZeros().toPlainString();
-        sb.append(' ').append(s).append(" seconds");
+        sb.append(s).append(" seconds ");
       }
     } else if (months == 0 && days == 0) {
-      sb.append(" 0 microseconds");
+      sb.append("0 microseconds ");
     }
 
+    sb.setLength(sb.length() - 1);
     return sb.toString();
   }
 
   private void appendUnit(StringBuilder sb, long value, String unit) {
     if (value != 0) {
-      sb.append(' ').append(value).append(' ').append(unit).append('s');
+      sb.append(value).append(' ').append(unit).append('s').append(' ');
```

Comment:
nit: Less `.append()` calls:
```suggestion
      sb.append(value).append(' ').append(unit).append("s ");
```